miniapp/miniapp.cpp

```diff
@@ -200,6 +204,26 @@ std::pair<cell_gid_type, cell_gid_type> distribute_cells(cell_size_type num_cell
     return {cell_from, cell_to};
 }
 
+std::vector<cell_gid_type> partition_domains(cell_size_type num_cells) {
+    using nest::mc::util::make_span;
+    using nest::mc::util::partition_functional;
+    
+    const auto num_domains = communication::global_policy::size();
+    const auto cells_per_domain = (cell_gid_type) (num_cells/(double)num_domains);
```

Comment:
Not round robin. Incrementally added and divided to have equal numbers on every node. So 0...4, 5...9, 10..14 on nodes 1, 2, and 3. The domain partition vector assumes that they're ordered, but not that they're equal sized for future, more sophisticated load balancing.